### DISCUSSION TITLE
Filtering grid using user data

### DISCUSSION THREAD

**Discussion by juliermefelix** [serenity-is/Serenity#6522] [Help] [answered]
I have the following scenario:
Actors: Teacher, Coordinator and Assistant
The teacher records the student's occurrence, this feeds the occurrence table with the student's data (enrollment, name, class and type of occurrence).
Each assistant is responsible for a class group (A, B, C, D, etc).
When registering the assistant in the system, I will include his classes in the user table.
The Assistant logs into the system and accesses the occurrence grid, at that moment I need only the classes corresponding to the assistant to be listed.
Today it is listing all the classes.
I'm using .net core 6
I tried without success this code in ListHandler.

` protected override void ApplyFilters(SqlQuery query)
        {
            var username = User?.Identity?.Name;

            base.ApplyFilters(query);

            if (UserRetriever.ByUsername(username) is UserDefinition user)
            {

                if (!Permissions.HasPermission("SomePermissionThatLetsUserSeeAllRows"))
                {
                    query.Where(fld.Fields.IdTurma == user.Turma);
                }
            }
        }
`
follow the images
Grid displaying all occurrences
![WhatsApp Image 2022-09-02 at 09 35 19](https://user-images.githubusercontent.com/7516443/188215513-1cdb103f-ddd2-44d4-91e1-46590336e27b.jpeg)

user registration
![WhatsApp Image 2022-09-02 at 09 36 20](https://user-images.githubusercontent.com/7516443/188215841-da3fa92e-3876-4671-ac04-504f1152c3df.jpeg)

**Comment by arunkumar12712** [comment, marked as answer]
var accessEmployee = 'retrive employee access of users'
query.Where(new Criteria(fld.IdTurma ).In(accessEmployee ));

**Reply by juliermefelix** [reply]
I managed to compare the user table with the occurrence table, but now it only displays the record of the last position.
needed the query to traverse the foreach
![image](https://user-images.githubusercontent.com/7516443/188496402-c87bfae0-b7b6-4b20-8fb3-3d3a773bd5e9.png)
it was supposed to be displaying four results
![image](https://user-images.githubusercontent.com/7516443/188496574-cc230ebf-ce23-4ffa-9bcf-075fe03f2956.png)

![image](https://user-images.githubusercontent.com/7516443/188496745-7cc3deb9-78cd-4d8e-874e-876b8f4c3c3e.png)

**Reply by arunkumar12712** [reply]
query.Where(new Criteria(MyRow.Field.Turma ).In(usuario.Turma.Split(',').ToArray()));
